packages/react/src/utils/sha1.ts

```diff
@@ -0,0 +1,170 @@
+type HashValueProps = {
+  sessionId: string | undefined | null;
+  userId: string | undefined | null;
+  orgId: string | undefined | null;
+};
+
+export const hashStateValues = (values: HashValueProps) => {
+  const sortedValues = Object.keys(values)
+    .sort()
+    .reduce((accumulator, key) => {
+      if (values[key as keyof HashValueProps]) {
+        // @ts-ignore
+        accumulator.push(values[key]);
+      }
+
+      return accumulator;
+    }, []);
+
+  if (!sortedValues.length) {
+    return null;
+  }
+
+  return SHA1(sortedValues.join('&')).slice(0, 8);
+};
+
+/////////
+///////
+// The SHA (Secure Hash Algorithm) can be used to encrypt data for secure transfer between applications.
+// The SHA1() function returns a string with the SHA1 encrypted hash as a string with 40 characters.
+// It is fully compatible with UTF-8 encoding.
+///
+//
+export const SHA1 = (msg: string) => {
```

Comment:
Good thinking extracting sha1 like this. I'm debating whether we need to go with SHA1 after all. 
Take a look at this: https://clerk.com/blog/generating-sortable-stripe-like-ids-with-segment-ksuids#clerk-s-id-generator  - Clerk is using [KSUID](https://github.com/segmentio/ksuid) to generate IDs

Given that our ID generation logic will likely always stay the same, would it be sufficient to use the existing ids to create a hash? I think we can extract part of each id:
 
![Screenshot 2023-07-18 at 12 36 07](https://github.com/clerkinc/javascript/assets/1811063/a9ac269a-187e-4c83-94ba-4bde096c1265)

and concatenate the parts together. If an id is undefined, we consider it to be  "" instead.

We're going to end up with signatures of around ~15chars long if orgs are enabled which is longer than the 8chars of the original design, but I think the tradeoff is worth it.

For the loading state where you have no ids available, we can probably default to `Math.random().toString(36).substr(2, 8);`

What do you think? @gkats @desiprisg

Reply:
@nikosdouvlis This makes sense to me to be honest

Reply:
It depends - I think implementing what I wrote above would be much simpler than embedding the SHA1 algorithm in our packages. 

A different approach would be to use https://developer.mozilla.org/en-US/docs/Web/API/SubtleCrypto/digest - did you explore that option? A small downside is that `digest` returns a Promise and we'd probably like `authStateKey` to be sync.

Reply:
@gkats Valid point. What if we used the [MD5](https://coursesweb.net/javascript/md5-hash-string_cs) hashing algorithm that is faster, instead of SHA1? 

Regarding cons(point 2) if we add something like organization role, for example, we can append that as a string since we don't care so much about security(e.4jdfjd8klj453kjh4admin)

Reply:
I think its safe to say that SubtleCrypto will be available in every platform where Clerk runs as far as web or cross-platform native apps are concerned. The digest method can produce a hash using the SHA1 algorithm so that wouldn't change across platforms anyway.
The biggest drawback of using the web crypto API directly is the fact that it is async and not sync as the original proposal. 

Regarding the performance of MD5 and SHA1, I think it doesn't matter in this case as the key will rarely change, both algorithms are fast enough.

What @gkats said about the size is true though, its going to go from 6-8chars to ~15 for our use case. We could pick less chars per id though as entropy should be enough regardless, I wouldn't worry too much about collisions or duplicate keys (false-negatives) on the frontend.

Reply:
Bundling sha1 definitely feels weird to me but it makes sense if you consider the above discussion so the decision is up to you!

If we decide to proceed with sha1 we should probably do the following refactors in this PR:
- separate the sha1 algo from the rest of the utils. Right now, the `sha1.ts` file has utils specific to `authStateKey`
- stop exporting sha1, export a `syncDigest` method instead

Reply:
@nikosdouvlis When went with sha1 and moved the functionality in Clerk-js.

Reply:
Folks have we considered returning the key from FAPI on the session object?